### PR TITLE
Desktop icons: add notification information

### PR DESCRIPTION
**Pull request by erikos**
When there are new notifications from the app (e.g. the ChatBox notifies the user regarding certain lessons) a badge is added to the desktop icon.

### CODE REVIEW

File: js/ui/appDisplay.js [hunk 1]
```diff
@@ -2382,7 +2401,10 @@ const AppIcon = new Lang.Class({
         return [sourceActor.actor];
     },
 
-    _onNewGtkNotificationSource: function(daemon, source) {
+    _sourceAdded: function(tray, source) {
+        if (!(source instanceof NotificationDaemon.GtkNotificationDaemonAppSource))
+            return;
```
Comment (smspillaz) [inline js/ui/appDisplay.js]: I don't know if it makes sense to have this limitation. What was the reason for not allowing Fdo notifications?

File: js/ui/notificationDaemon.js [hunk 2]
```diff
@@ -762,6 +764,10 @@ const GtkNotificationDaemonAppSource = new Lang.Class({
         }
         return [this._appId, notifications];
     },
+
+    get app() {
+        return this._app;
+    }
```
Comment (smspillaz) [inline js/ui/notificationDaemon.js]: Ah, got it, this is used in

    if (source.app != this.app)
              return;

File: js/ui/appDisplay.js (hunk 1 above)
Comment (erikos) [inline js/ui/appDisplay.js]: Well, we only connect to the signal for the ChatBox case. So I want to filter and bail out as early as possible. If we open it up for more apps we can remove that special casing.

File: js/ui/notificationDaemon.js (hunk 2 above)
Comment (erikos) [inline js/ui/notificationDaemon.js]: Right, that is needed for this check.

File: js/ui/appDisplay.js (hunk 1 above)
Comment (smspillaz) [inline js/ui/appDisplay.js]: Hmm okay. Can you add a comment specifying this? It might be confusing for future maintainers if they see this.

Comment (erikos) [inline js/ui/appDisplay.js]: I added a comment for that one.

**Pull request merged**